cpg-language-jvm/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/jvm/ExpressionHandler.kt

```diff
@@ -0,0 +1,376 @@
+ * limitations under the License.
+ *
+ *                    $$$$$$\  $$$$$$$\   $$$$$$\
+ *                   $$  __$$\ $$  __$$\ $$  __$$\
+ *                   $$ /  \__|$$ |  $$ |$$ /  \__|
+ *                   $$ |      $$$$$$$  |$$ |$$$$\
+ *                   $$ |      $$  ____/ $$ |\_$$ |
+ *                   $$ |  $$\ $$ |      $$ |  $$ |
+ *                   \$$$$$   |$$ |      \$$$$$   |
+ *                    \______/ \__|       \______/
+ *
+ */
+package de.fraunhofer.aisec.cpg.frontends.jvm
+
+import de.fraunhofer.aisec.cpg.frontends.Handler
+import de.fraunhofer.aisec.cpg.graph.*
+import de.fraunhofer.aisec.cpg.graph.statements.expressions.*
+import de.fraunhofer.aisec.cpg.graph.types.FunctionType
+import de.fraunhofer.aisec.cpg.passes.SymbolResolver
+import sootup.core.jimple.basic.Local
+import sootup.core.jimple.basic.Value
+import sootup.core.jimple.common.constant.*
+import sootup.core.jimple.common.expr.*
+import sootup.core.jimple.common.ref.*
+import sootup.core.signatures.MethodSignature
+import sootup.core.signatures.SootClassMemberSignature
+import sootup.java.core.jimple.basic.JavaLocal
+
+class ExpressionHandler(frontend: JVMLanguageFrontend) :
+    Handler<Expression, Value, JVMLanguageFrontend>(
+        ::ProblemExpression,
+        frontend,
+    ) {
+
+    init {
+        map.put(Local::class.java) { handleLocal(it as Local) }
+        map.put(JavaLocal::class.java) { handleLocal(it as Local) }
+        map.put(JThisRef::class.java) { handleThisRef(it as JThisRef) }
+        map.put(JParameterRef::class.java) { handleParameterRef(it as JParameterRef) }
+        map.put(JInstanceFieldRef::class.java) { handleInstanceFieldRef(it as JInstanceFieldRef) }
+        map.put(JStaticFieldRef::class.java) { handleStaticFieldRef(it as JStaticFieldRef) }
+        map.put(JArrayRef::class.java) { handleArrayRef(it as JArrayRef) }
+        map.put(JInterfaceInvokeExpr::class.java) {
+            handleInterfaceInvokeExpr(it as JInterfaceInvokeExpr)
+        }
+        map.put(JVirtualInvokeExpr::class.java) {
+            handleVirtualInvokeExpr(it as JVirtualInvokeExpr)
+        }
+        map.put(JDynamicInvokeExpr::class.java) {
+            handleDynamicInvokeExpr(it as JDynamicInvokeExpr)
+        }
+        map.put(JSpecialInvokeExpr::class.java) { handleSpecialInvoke(it as JSpecialInvokeExpr) }
+        map.put(JStaticInvokeExpr::class.java) { handleStaticInvoke(it as JStaticInvokeExpr) }
+        map.put(JNewExpr::class.java) { handleNewExpr(it as JNewExpr) }
+        map.put(JNewArrayExpr::class.java) { handleNewArrayExpr(it as JNewArrayExpr) }
+        map.put(JNewMultiArrayExpr::class.java) {
+            handleNewMultiArrayExpr(it as JNewMultiArrayExpr)
+        }
+        map.put(JCastExpr::class.java) { handleCastExpr(it as JCastExpr) }
+
+        // Binary operators
+        // - Equality checks
+        map.put(JEqExpr::class.java) { handleAbstractBinopExpr(it as AbstractBinopExpr) }
```

Comment:
I assume there is no common superclass that is more specific than `AbstractBinopExpr` and there is a subclass of AbstractBinopExpr that has to be handled differently?

Should we add a fallthrough default case that catches the remaining `AbstractBinopExpr`?

Reply:
I added it just to be sure